sycl/include/CL/sycl/xilinx/fpga/memory_properties.hpp

```diff
@@ -0,0 +1,55 @@
+//==- memory_properties.hpp --- SYCL Xilinx memory proprerties       -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SYCL_XILINX_FPGA_MEMORY_PROPERTIES_HPP
+#define SYCL_XILINX_FPGA_MEMORY_PROPERTIES_HPP
+
+#include "CL/sycl/detail/defines.hpp"
+#include "CL/sycl/detail/property_helper.hpp"
+#include "CL/sycl/properties/accessor_properties.hpp"
+
+__SYCL_INLINE_NAMESPACE(cl) {
+
+namespace sycl {
+namespace xilinx {
+namespace property {
+
+struct ddr_bank {
```

Comment:
This could be a higher-order self-describing property to be introspected by Clang to do what is required to be done...